internal/config/config.go

```diff
@@ -121,6 +102,7 @@ func (c *Config) Load(ctx context.Context) error {
 	}
 
 	authClient.Timeout = c.requestTimeout()
+	//nolint:staticcheck // We should move to subsystem loggers, but that is a much bigger change
```

Comment:
Punted on this to avoid setting up subsystem loggers, which requires adding new subsystems to `ctx` and may also require defining new environment variables to configure log level per subsystem.

We could also consider just using [NewLoggingHTTPTransport](https://developer.hashicorp.com/terraform/plugin/sdkv2/logging/http-transport#newlogginghttptransport-transport-http-roundtripper) for now; that would avoid the need to add new subsystems to the context on provider startup but we would lose the log tagging we have now (e.g., at the moment packngo logs are logged with an `Equinix Metal (packngo)` prefix).